### PR TITLE
Add `QualifiedType::isNonPointerScalar()`

### PR DESCRIPTION
**Pull request by AlviseDeFaveri**

### CODE REVIEW

File: include/revng/Model/QualifiedType.h [hunk 1]
```diff
@@ -51,9 +51,14 @@ class model::QualifiedType : public model::generated::QualifiedType {
...
   bool isArray() const;
-  /// Checks if is a pointer type, withouth unwrapping typedefs
+  /// Checks if is a pointer type, unwrapping typedefs
   bool isPointer() const;
 
+  /// Checks if it contains a scalar type and is not a pointer nor an array
+  bool isNonPointerScalar() const {
+    return isScalar() and not(isPointer() or isArray());
+  }
+
```
[the remark this answers is not in the document]
Comment (AlviseDeFaveri) [reply include/revng/Model/QualifiedType.h:61]: I don't have strong opinions on this, there are times when you want to check if a value is just a (possibly const?) plain "primitive type" which is trivially convertible to/from LLVM IR without additional thinking.

Here's the related `revng-c` thread https://rev.ng/gitlab/revng-private/revng-c/-/merge_requests/80#note_15917

[the remark this answers is not in the document]
Comment (aleclearmind) [reply include/revng/Model/QualifiedType.h:61]: We can have `isPrimitive`! Maybe we can have `is(TypeKind)` so we can have `is(StructKind)` too.

[the remark this answers is not in the document]
Comment (AlviseDeFaveri) [reply include/revng/Model/QualifiedType.h:61]: yeah, but it should be "just" primitive (no pointer or array qualifiers). If you don't see a general use case for that, I'll go back to using this privately in the module.

[the remark this answers is not in the document]
Comment (AlviseDeFaveri) [reply include/revng/Model/QualifiedType.h]: @aleclearmind please check if the new version corresponds to what you meant

[the remark this answers is not in the document]
Comment (aleclearmind) [reply include/revng/Model/QualifiedType.h]: Actually, we already `isPrimitive`.
I'd like to have a unified way to do this. Right now, in `is(...)` you have `not(isPointer() or isArray)` but I think what you want to say is "only return true if there are no qualifiers or there are only `const` qualifiers". Take a look at `isPrimitive`, I think that's a more appropriate approach.

**Pull request merged**